resource_handler_test.go

```diff
@@ -27,6 +27,36 @@ type testResourceHandler struct {
 	data map[string]testData
 }
 
+func (h testResourceHandler) shouldReturnNoContent(id string, op PatchOperation) bool {
```

Comment:
This is looking much better @kskumgk63! Good job.

There is one more thing we need to take into account when testing this:
1. The [`add` operation](https://tools.ietf.org/html/rfc7644#section-3.5.2.1) should return `No Content` only when the value **already exists** AND is the **same**.

2. The [`remove` operation](https://tools.ietf.org/html/rfc7644#section-3.5.2.2) should return `No Content` when the value to be remove is **already absent**.

3. (The [`replace` operation](https://tools.ietf.org/html/rfc7644#section-3.5.2.3) works the same as the `add` operation as far as I know.)
---

Small example:

```
dataValue, ok := h.data[id]
if !ok {
    return false
}
```
- 🆗 | Adding a non existing value returns `false`.
- 🚫 | Removing a non existing value returns `false`. Should be `true`.

Reply:
Covert the remove case!
https://github.com/elimity-com/scim/pull/102/commits/bd01f3b5f3fa0a0c17ae8a9179050fdfac8254ed